File: remove-nth-node-from-end-of-list.php
<?php
/**
 * Definition for a singly-linked list.
 */
 
 class ListNode {
    public $val = 0;
    public $next = null;
    function __construct($val = 0, $next = null) {
        $this->val = $val;
        $this->next = $next;
    }
}

class Solution {
    var $head;
    function __construct()
    {
        $this->head = null;
    }

    function push($data)
    {
        $new_node = new ListNode();
        $new_node->next = $this->head;
        $new_node->val = $data;
        $this->head = $new_node;
    }

    function printList()
    {
        $tmp = $this->head;
        $k = 0;
        while($tmp)
        {
            echo $tmp->val.' ';
            $tmp = $tmp->next;
        }
    }
    /**
     * @param ListNode $head
     * @param Integer $n
     * @return ListNode
     */
    function removeNthFromEnd($head, $n) {
        $fast_p = $head;
        $slow_p = $head;

        foreach( range(0, $n-1) as $k => $a ) {
            if( $fast_p->next == null ) {
                if( $k == $n - 1 ) {
                    $head = $head->next;
                }
                return $head;
            }
            $fast_p = $fast_p->next;
        }

        while( $fast_p->next != null ) {
            $fast_p = $fast_p->next;
            $slow_p = $slow_p->next;
        }

        if( $slow_p->next != null ) {
            $slow_p->next = $slow_p->next->next;
        }

        return $head;
    }
}

$sol = new Solution();
// $sol->push(5);
// $sol->push(4);
// $sol->push(3);
$sol->push(2);
$sol->push(1);
$sol->printList();
echo "\nhere:";
$d = $sol->removeNthFromEnd($sol->head, 1);
var_dump($d);

File: remove-nth-node-from-end-of-list.py
from typing import Optional

# Definition for singly-linked list.
class ListNode:
    def __init__(self, val=0, next=None):
        self.val = val
        self.next = next

class Solution:
    def __init__(self):
        self.head = None

    def push(self, data):
        new_node = ListNode()
        new_node.val = data
        new_node.next = self.head
        self.head = new_node

    def printList(self):
        tmp = self.head
        while(tmp):
            print(tmp.val, end=" ")
            tmp = tmp.next

    def removeNthFromEnd(self, head: Optional[ListNode], n: int) -> Optional[ListNode]:
        slow_p = head
        fast_p = head

        for i in range(n):
            if fast_p.next is None:
                if i == n - 1:
                    head = head.next
                return head
            fast_p = fast_p.next

        while( fast_p.next != None ):
            slow_p = slow_p.next
            fast_p = fast_p.next

        if slow_p.next is not None:
            slow_p.next = slow_p.next.next

        return head

# list1 = Solution()
# list1.push(5)
# list1.push(4)
# list1.push(3)
# list1.push(2)
# list1.push(1)

list1 = Solution()
list1.push(1)

print("Linked List ")
list1.printList()
print('')

d = list1.removeNthFromEnd(list1.head, 1)

print("Linked List after")
print(d)